src/features/generic_events/instrument/index.js

```diff
@@ -7,22 +7,17 @@ import { handle } from '../../../common/event-emitter/handle'
 import { windowAddEventListener } from '../../../common/event-listener/event-listener-opts'
 import { InstrumentBase } from '../../utils/instrument-base'
 import { FEATURE_NAME, OBSERVED_EVENTS, OBSERVED_WINDOW_EVENTS } from '../constants'
+import { getEnabledDatasources } from '../shared/utils'
 
 export class Instrument extends InstrumentBase {
   static featureName = FEATURE_NAME
   constructor (agentRef, auto = true) {
     super(agentRef, FEATURE_NAME, auto)
-    /** config values that gate whether the generic events aggregator should be imported at all */
-    const genericEventSourceConfigs = [
-      agentRef.init.page_action.enabled,
-      agentRef.init.performance.capture_marks,
-      agentRef.init.performance.capture_measures,
-      agentRef.init.user_actions.enabled,
-      agentRef.init.performance.resources.enabled
-    ]
+
+    const { datasources, anyEnabled } = getEnabledDatasources(this.agentRef.init)
```

Comment:
I like the introduction of the utils function to help manage the logic.  I am wondering if this is also introducing a shift in the usage pattern (very superficially).  Would it be possible to keep the logic "hidden" within the model (it looks like we're mostly checking against `init.feature_flags`)?  I also find it easier to grok if `enabled` stays the source of truth

Reply:
this is a good idea.  I'll hit you on slack and maybe you can expound on that a bit more